_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,199 @@
+
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // Send messages to Kafka
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    // Deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // Deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
+}
+```  
+
+This test case as you can ensuring that we are able to receive messages from `KafkaBridge` component. In the setup phase 
+we deploy `cluster operator`, one node `Kafka cluster` and also the `KafkaBridge` itself. Everything is inside the real
+Kubernetes cluster. All test cases with related test suites you can find in the `system module` to make it more sense. What 
+is also worth to mentioned that our system test currently takes around ~30 hours but we have 3 separate sub-sets, which 
+runs in parallel so it is "only" approximately 10h per each sub-set.
+
+## How they run with relation with Kubernetes
+
+Basically, what I mentioned in previous section system tests run against real Kubernetes cluster. You can run these 
+tests against `Minikube`, big `Kubernetes` or `Openshift` cluster. In the `test module` we have our representation
+of cluster which decide on which environment it is running. Moreover, we are using `azure pipelines` to trigger these
+tests and using `Minikube`.
+
+## How to run them
+
+Running the system tests is easy thing and nobody should be scared of. The steps are as follows:
+
+1. Fork the `strimzi-kafka-operator` repository - https://github.com/strimzi/strimzi-kafka-operator
+2. Build the whole project in root directory - `mvn clean install -DskipTests`
+3. Setup your Kubernetes cluster
+4. Login to your Kubernetes cluster
+5. Try to start some of our tests inside system test module for instance - `io.strimzi.systemtest.bridge.HttpBridgeST.testSendSimpleMessage()` (take takes around 5 minutes)
```

Comment:
It is important to explain how to run them with custom images since that is what many people might need to know. Also, you can mention that you can just run them from IDE.

Reply:
Yes good point.